File: src/main/java/no/nav/pensjon/regler/domain/beregning/penobjekter/BeregningYtelseskomponent.java
package no.nav.pensjon.regler.domain.beregning.penobjekter;

import no.nav.pensjon.regler.domain.beregning.Beregning;
import no.nav.pensjon.regler.domain.beregning.Ytelseskomponent;

/**
 * Created by N123422 on 06.11.2014.
 */
public abstract class BeregningYtelseskomponent extends Ytelseskomponent {

    private static final long serialVersionUID = -3590463254151748326L;

    //Benyttes i PEN domenemodell for MapKey in Beregning.
    private String ytelseKomponentTypeName;

    private Beregning beregning;

    public BeregningYtelseskomponent() {
        super();
    }

    public BeregningYtelseskomponent(BeregningYtelseskomponent beregningYtelseskomponent) {
        super(beregningYtelseskomponent);
        ytelseKomponentTypeName = beregningYtelseskomponent.ytelseKomponentTypeName;
    }

    protected Beregning getBeregning() {
        return beregning;
    }

    protected void setBeregning(Beregning beregning) {
        this.beregning = beregning;
    }
}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning/penobjekter/BeregningYtelseskomponent.kt
package no.nav.pensjon.regler.domain.beregning.penobjekter

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo
import no.nav.pensjon.regler.domain.beregning.Beregning
import no.nav.pensjon.regler.domain.beregning.Ytelseskomponent

/**
 * Created by N123422 on 06.11.2014.
 */
@JsonSubTypes(
    JsonSubTypes.Type(value = MotregningYtelseskomponent::class),
    JsonSubTypes.Type(value = SkattefriGrunnpensjon::class),
    JsonSubTypes.Type(value = SkattefriUforetrygdOrdiner::class)
)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
abstract class BeregningYtelseskomponent : Ytelseskomponent() {
    //Benyttes i PEN domenemodell for MapKey in Beregning.
    private var ytelseKomponentTypeName: String? = null
    protected var beregning: Beregning? = null
}